### ISSUE TITLE
Support Slack Ephemeral Messages

### ISSUE THREAD

**Issue by mikegrima** [Netflix/hubcommander#56]
Need to add support for Slack's ephemeral messages for more informational -- not super important for everyone else type of messages.

This could also prove useful for dealing with Duo auth messages.

**Comment by tezansahu** [comment]
I wish to work on this issue. Pls let me know the details

**Comment by mikegrima** [comment]
Hello @tezansahu !

Thank you for looking into this :)

The long and the short of this is that it should be implemented as a dictionary parameter in the `say_*` methods that I have. Something like `ephemeral=True`.  That should follow the Slack API so that ephemeral messages are sent.

This way, anyone who implements a plugin can write out ephemeral messages.

**Comment by viveksyngh** [comment]
is anybody working on this one ?

**Comment by mikegrima** [comment]
Hello @viveksyngh . If @tezansahu isn't working on it, you are more than welcome to.

It's not on my timeline for Q4 (maybe Q1).

**Comment by viveksyngh** [comment]
Okay .. I will work on this. to send an ephemeral slack api also required user id ? How that user id will be passed ?

**Comment by mikegrima** [comment]
We currently make use of Slack's python-rtmbot which itself makes use of the [python-slackclient](https://github.com/slackapi/python-slackclient).

The `slackclient` is set up [here](https://github.com/Netflix/hubcommander/blob/develop/hubcommander.py#L91-L92).  Once's that is setup, it is then placed into the `bot_components` package and referenced in `slack_comm.py`.  [Here is an example](https://github.com/Netflix/hubcommander/blob/develop/bot_components/slack_comm.py#L26-L27).

It looks like https://github.com/slackapi/python-slackclient/blob/2a9c54601b98ce4559f4fd8dd30635ec248c101c/docs-src/basic_usage.rst#sending-a-message documents how to send an ephemeral message via the `slackclient`.

**Comment by viveksyngh** [comment]
Thanks you explanation and references. As per python-slackclient documentation 
_Sending an ephemeral message, which is only visible to an assigned user in a specified channel, is nearly the same as sending a regular message, but with an additional user parameter._

f```
rom slackclient import SlackClient

slack_token = os.environ["SLACK_API_TOKEN"]
sc = SlackClient(slack_token)

sc.api_call(
  "chat.postEphemeral",
  channel="#python",
  text="Hello from Python! :tada:",
  user="U0XXXXXXX"
)
```
so will this user will also be an argument of **say** which will none for other case but some value in case of empheral message ?

**Comment by mikegrima** [comment]
I think it makes sense to have a keyword arg for `ephemeral_user=THEUSER` to all of the `say` functions.

The user details are a `dict` that is mostly passed into each command via the `get_user_data` function in `slack_comm.py`: https://github.com/Netflix/hubcommander/blob/develop/bot_components/slack_comm.py#L110

This shows the command being executed and getting `user_data` passed in:
https://github.com/Netflix/hubcommander/blob/develop/hubcommander.py#L74-L75

**Comment by viveksyngh** [comment]
Should not functions like ```send_error```, ```send_info```, ```send_success``` should be modified to send ```ephemeral``` message and ```ephemeral_user``` values ?

**Comment by mikegrima** [comment]
Implemented in #63
